xcdat/regridder/regrid2.py

```diff
@@ -0,0 +1,547 @@
+from typing import Any, Dict, List, Tuple
+
+import numpy as np
+import xarray as xr
+
+from xcdat.regridder.base import BaseRegridder
+
+
+def extract_bounds(bounds: xr.DataArray) -> Tuple[xr.DataArray, xr.DataArray]:
+    """
+     Extract lower and upper bounds from an axis.
+
+     Parameters
+     ----------
+     bounds : xr.DataArray
+         Dataset containing axis with bounds.
+
+     Returns
+     -------
+    Tuple[xr.DataArray, xr.DataArray]
+         A tuple containing the lower and upper bounds for the axis.
+    """
+    if bounds[0, 0] < bounds[0, 1]:
+        lower = bounds[:, 0]
+        upper = bounds[:, 1]
+    else:
+        lower = bounds[:, 1]
+        upper = bounds[:, 0]
+
+    return lower, upper
+
+
+def map_latitude(src: xr.DataArray, dst: xr.DataArray) -> Tuple[List, List]:
+    """
+    Map source to destination latitude.
```

Comment:
```suggestion
    """Map source to destination latitude.
```

Add description in docstring on how the algorithm/logic works